From cdad0e3324a82ac12e8934d00d7e9d0c288ff037 Mon Sep 17 00:00:00 2001
From: DevonTM <ipangcheater@gmail.com>
Date: Thu, 12 Mar 2026 14:40:53 +0700
Subject: [PATCH] Remove #undef SIZE_MAX

Remove obsolete unconditional #undef SIZE_MAX from MinGW headers (leftover from old gettext); breaks mingw-w64 GCC with MCF thread model.
---
 src/gai_strerror.h | 4 ----
 src/getaddrinfo.h  | 4 ----
 src/gettimeofday.h | 4 ----
 3 files changed, 12 deletions(-)

diff --git a/src/gai_strerror.h b/src/gai_strerror.h
index da7b6ca7..1ee77ca2 100644
--- a/src/gai_strerror.h
+++ b/src/gai_strerror.h
@@ -33,10 +33,6 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#ifdef __MINGW32__
-#  undef SIZE_MAX
-#endif // __MINGW32__
-
 #ifndef EAI_SYSTEM
 #  define EAI_SYSTEM -11 /* System error returned in `errno'.  */
 #endif
diff --git a/src/getaddrinfo.h b/src/getaddrinfo.h
index d841d4be..e6b9dbea 100644
--- a/src/getaddrinfo.h
+++ b/src/getaddrinfo.h
@@ -33,10 +33,6 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#ifdef __MINGW32__
-#  undef SIZE_MAX
-#endif // __MINGW32__
-
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif // HAVE_CONFIG_H
diff --git a/src/gettimeofday.h b/src/gettimeofday.h
index c5231e9f..6eb74f77 100644
--- a/src/gettimeofday.h
+++ b/src/gettimeofday.h
@@ -36,10 +36,6 @@
 #ifndef _D_GETTIMEOFDAY_H
 #define _D_GETTIMEOFDAY_H 1
 
-#ifdef __MINGW32__
-#  undef SIZE_MAX
-#endif // __MINGW32__
-
 #ifdef HAVE_CONFIG_H
 #  include "config.h"
 #endif // HAVE_CONFIG_H
-- 
2.49.0.windows.1

